### DISCUSSION TITLE
Seeder in Migration

### DISCUSSION THREAD

**Discussion by exptjs** [mikro-orm/mikro-orm#6225] [Q&A]
Please tell me is it possible to use `Seeder` in `Migration`?
```
import {Seeder} from '@mikro-orm/seeder';
import {EntityManager} from '@mikro-orm/core';
import {user_roles} from '../entities/user_roles';

export class Migration20241109104313 extends Seeder {
  async run(em: EntityManager): Promise<void> {
    const userRoleAdmin = em.create(user_roles, {
      code_id: 1,
      name: 'admin',
      created_at: new Date(),
      updated_at: new Date()
    });

    const userRoleUser = em.create(user_roles, {
      code_id: 2,
      name: 'user',
      created_at: new Date(),
      updated_at: new Date()
    });
  }
}
```
`MigrationError: Migration Migration20241109104313 (up) failed: Original error: migration.reset is not a function`

**Comment by B4nan** [comment]
Technically yes, but it will be error prone, I wouldn’t suggest anyone doing this, neither using the entity manager. Migrations should be static, if you use entities and modify them, you will break things. Imagine you add a new property at some point, but your old migration which uses the seeder will now use the new entity definition. It will generate queries with this new column which wont be present in your schema just yet.

**Reply by B4nan** [reply]
Looking at what you did there, that is clearly wrong, you cant just extend the seeder class instead of migration class. You would need to create a seeder as usual, and create the seeder manager by hand inside your migration, which you would use as described in the docs. 

But as described above, this can break over time and is not a good idea in the first place.

**Reply by exptjs** [reply]
If this is possible, can you tell me why the error occurs? I would like to implement this through migrations.

**Reply by B4nan** [reply]
I already told you in the comment above.
